BUILD.gn

```diff
@@ -611,20 +610,52 @@ if (is_mac) {
   }
 }
 
+cargo_audit_target_dir = "$root_build_dir/cargo_audit"
+cargo_audit_exe = "$cargo_audit_target_dir/release/cargo-audit"
+cargo_audit_src = "//brave/tools/crates/cargo_audit/v0_17/crate"
+if (host_os == "win") {
+  cargo_audit_exe += ".exe"
+}
+
+action("build_cargo_audit") {
+  script = "//build/gn_run_binary.py"
+
+  cargo_exe = "//third_party/rust-toolchain/bin/cargo"
+  if (host_os == "win") {
+    cargo_exe += ".exe"
+  }
+
+  inputs = [
+    cargo_exe,
+    "$cargo_audit_src/Cargo.lock",
+  ]
+  outputs = [ cargo_audit_exe ]
+  args = [
+    rebase_path(cargo_exe),
+    "build",
+    "--release",
```

Comment:
How are we feeling about `--locked`/`--frozen` this week?

Reply:
Good question, I just used the same command that upstream recommended for `gnrt`. I guess this could have the same issues as we had with the old download_rust_deps script so maybe we should use locked here?

Reply:
Seems like each have their own failure mode. I'm fine with it either way.